valohai/internals/files.py

```diff
@@ -16,16 +16,36 @@ def get_glob_pattern(source: str) -> str:
     return source
 
 
-def expand_globs(sources: Union[str, list], preprocessor=lambda s: s) -> Set[str]:
+def expand_globs(
+    sources: Union[str, list], preprocessor=lambda s: s
+) -> Set[Tuple[str, str]]:
+    """Returns a set of paths as a result of expanding all the source wildcards
+
+    First item of the resulting tuple item is the expanded path
+    Second item of the resulting tuple is the logical root path
+
+    Example
+    source: /tmp/foo/**/*.png
+    output: set(("/tmp/foo/bar/hello.png", "/tmp/foo"), ("/tmp/foo/bar/sub/jello.png", "/tmp/foo"))
+
+    source: /tmp/yeah/*.png
+    output: set(("/tmp/yeah/hello.png", "/tmp/yeah"), ("/tmp/yeah/jello.png", "/tmp/yeah"))
+
+    :param sources: Path or list of paths. May contain * or ** wildcards.
+    :param preprocessor: Preprocessor to be used for each of the paths
+    :return:
+    """
+
     if isinstance(sources, str):
         sources = [sources]
     files_to_compress = set()
     for source_path in sources:
         source_path = preprocessor(source_path)
         glob_pattern = get_glob_pattern(source_path)
-        for file_path in glob.glob(glob_pattern):
+        root_path = os.path.dirname(glob_pattern).split("*")[0]  # Handles ** also
```

Comment:
Only one split needed here. 
```suggestion
        root_path = os.path.dirname(glob_pattern).split("*", 1)[0]  # Handles ** also
```
You could also use `partition`: 
```suggestion
        root_path = os.path.dirname(glob_pattern).partition("*")[0]  # Handles ** also
```